base/utils/R/convert.input.R

```diff
@@ -10,7 +10,7 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
                           end_date, pkg, fcn, con = con, host, browndog, write = TRUE, 
                           format.vars, overwrite = FALSE, exact.dates = FALSE, 
                           allow.conflicting.dates = TRUE, insert.new.file = FALSE, pattern = NULL,
-                          forecast = FALSE, ensemble = FALSE,...) {
+                          forecast = FALSE, ensemble = FALSE, ensemble_name = NULL, ...) {
```

Comment:
Undocumented new argument. What is this and do we really need it? Looks like we're just using it as `paste(filename_pattern, ensemble_name, sep="\\.")` which suggests to me that it's just simpler to ask the function calling this to do that paste when it creates the pattern it's looking for.

Reply:
That is what it's used for.  The issue is that sometimes convert.input handles all ensembles at once (during the raw data download) and sometimes it handles individual ensemble members (during gapfilling and met2model processing), where it's being called iteratively by met.process.  In the latter case, convert.input is not aware of which ensemble member it's trying to process, and needs that information to properly check if the file already exists in the database.

If I put this code in `metgapfill.module` and `met2model.module`, convert.input wouln't be able to properly build its regular expressions when processing ensembles iteratevely in the download stage.  convert.input is only called once in the download stage; download.NOAA_GEFS generates the entire ensemble of files at once, which need to be iterated over.  convert.input wouldn't be able to build its regular expression because the code for that would have been moved to outside the function.

In short, the problem is that the iteration is happening in two different places: sometimes in convert.input and sometimes in met.process.

The regular expression thing is a workaround the fact that the inputs table doesn't have a column for ensemble, so it's impossible to uniquely identify by column content alone.  The only place where ensemble identification is found is in the file name, (and folder name), which is why a regex is used to match the appropriate file name.  Even if an ensemble column did exist, however, convert.input would still need the information about which ensemble it's trying to process.

None of the arguments to convert.input have documentation on top of the function header.  I can write some.  Where else am I supposed to add documentation for new arguments?